master/internal/db/postgres_trial.go

```diff
@@ -164,22 +164,32 @@ func (db *PgDB) fullTrialSummaryMetricsRecompute(
 	ctx context.Context, tx *sqlx.Tx, trialID int,
 ) error {
 	trainSummary, err := db.calculateFullTrialSummaryMetrics(
-		ctx, tx, trialID, "avg_metrics", TrainingMetric)
+		ctx, tx, trialID, TrainingMetric)
 	if err != nil {
 		return fmt.Errorf("rollback computing training summary metrics: %w", err)
 	}
 	valSummary, err := db.calculateFullTrialSummaryMetrics(
-		ctx, tx, trialID, "validation_metrics", ValidationMetric)
+		ctx, tx, trialID, ValidationMetric)
+	if err != nil {
+		return fmt.Errorf("rollback computing validation summary metrics: %w", err)
+	}
+	genericSummary, err := db.calculateFullTrialSummaryMetrics(
+		ctx, tx, trialID, GenericMetric)
 	if err != nil {
 		return fmt.Errorf("rollback computing validation summary metrics: %w", err)
 	}
 
 	updatedSummaryMetrics := model.JSONObj{}
 	if len(trainSummary) > 0 {
-		updatedSummaryMetrics["avg_metrics"] = trainSummary
+		key := model.TrialSummaryMetricsJSONPath(model.TrainingMetricType)
+		updatedSummaryMetrics[key] = trainSummary
 	}
 	if len(valSummary) > 0 {
-		updatedSummaryMetrics["validation_metrics"] = valSummary
+		key := model.TrialSummaryMetricsJSONPath(model.ValidationMetricType)
+		updatedSummaryMetrics[key] = valSummary
+	}
+	if len(genericSummary) > 0 {
+		updatedSummaryMetrics["generic_metrics"] = genericSummary
```

Comment:
I thought that the json path for the summary should include the user-specified `metricType`, perhaps literally be the `metricType`.
e.g. say user reports metric of type `inference`, name `loss`, and then another metric of type `potato`, also name `loss`. with this, it seems that they'd be summarized as one?